executable/executable.go

```diff
@@ -30,20 +30,18 @@ type Executable struct {
 
 	Process *os.Process
 
-	StdinPipe io.WriteCloser
-
 	// These are set & removed together
 	atleastOneReadDone bool
 	cmd                *exec.Cmd
-	stdoutPipe         io.ReadCloser
-	stderrPipe         io.ReadCloser
 	stdoutBytes        []byte
 	stderrBytes        []byte
 	stdoutBuffer       *bytes.Buffer
 	stderrBuffer       *bytes.Buffer
 	stdoutLineWriter   *linewriter.LineWriter
 	stderrLineWriter   *linewriter.LineWriter
 	readDone           chan bool
+
+	stdioHandler stdioHandler
```

Comment:
Club this with the values that are set & removed together - they should be set in Start, removed in Wait and no way to meddle with them in between.